sycl/source/detail/global_handler.cpp

```diff
@@ -53,12 +53,10 @@ class ObjectUsageCounter {
       return;
 
     MCounter--;
-    if (!MCounter) {
-      LockGuard Guard(GlobalHandler::MSyclGlobalHandlerProtector);
-      GlobalHandler *RTGlobalObjHandler = GlobalHandler::getInstancePtr();
-      if (RTGlobalObjHandler) {
-        RTGlobalObjHandler->prepareSchedulerToRelease();
-      }
+    LockGuard Guard(GlobalHandler::MSyclGlobalHandlerProtector);
+    GlobalHandler *RTGlobalObjHandler = GlobalHandler::getInstancePtr();
+    if (RTGlobalObjHandler) {
+      RTGlobalObjHandler->prepareSchedulerToRelease(!MCounter);
```

Comment:
Should we put `MCounter-- ` under the Guard? If MCounter is decremented by two threads approximately at the same time then we will have MCounter == 0 for two threads, so we will call RTGlobalObjHandler->prepareSchedulerToRelease(true) twice. Is this a mistake or is it safe to call prepareSchedulerToRelease(true) twice? Probably I am missing something.

Reply:
it should not be a problem but I would prefer to fix this. I haven't noticed that. Thank you!

Reply:
fixed here https://github.com/intel/llvm/pull/11119/commits/52a15f46cf33b16b917b6cc3c3a69b21ae235006